### PR TITLE
TimeZone: added E2E-tests to verify the custom time zone support.

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
These tests verifies the functionality of the custom time zones support.

**Special notes for your reviewer**:
Had a lot of trouble getting the iterator of several panels to work. Needed to dig into how our e2e abstraction work and also how cypress runtime is working. But in the end it turned out ok.

### CODE REVIEW

**Comment by hugohaggmark** [comment]
Would love to hear more about the challenges you had @mckn and if you have general advice then maybe you could add some details here https://github.com/grafana/grafana/blob/master/contribute/style-guides/e2e.md

File: packages/grafana-e2e-selectors/src/selectors/components.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export const Components = {
     Panel: {
       title: (title: string) => `Panel header title item ${title}`,
       headerItems: (item: string) => `Panel header item ${item}`,
+      containerByTitle: (title: string) => `Panel container title ${title}`,
```
Comment (hugohaggmark) [inline packages/grafana-e2e-selectors/src/selectors/components.ts:21]: Out of curiosity, what's the difference between `title` and `containerByTitle `? I think might need to remove one of them.

File: e2e/suite1/specs/dashboard-time-zone.spec.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            .labels()
+            .should('be.visible')
+            .last()
+            .should(element => {
+              timesInUtc[title] = element.text();
+            })
+        );
+    }
+
+    e2e.pages.Dashboard.Toolbar.toolbarItems('Dashboard settings').click();
```
Comment (hugohaggmark) [inline e2e/suite1/specs/dashboard-time-zone.spec.ts:41]: nit: I just had an idea, this should really be this so we don't expose magic strings. But I can fix that in another PR.
```suggestion
    e2e.pages.Dashboard.Toolbar.dashboardSettings().click();
```

**Review by hugohaggmark — APPROVED** [review]
Looks great! Left just small comments for you to consider, let me know if they're unclear.

File: packages/grafana-e2e-selectors/src/selectors/components.ts (hunk 1 above)
Comment (mckn) [reply packages/grafana-e2e-selectors/src/selectors/components.ts:21]: Container is targeting the panel container so you can access its children in an easy way. The Title is targeting the actual title/header.

File: e2e/suite1/specs/dashboard-time-zone.spec.ts (hunk 2 above)
Comment (mckn) [reply e2e/suite1/specs/dashboard-time-zone.spec.ts:41]: ah, nice! I can update that one.

**Comment by mckn** [comment]
> Would love to hear more about the challenges you had @mckn and if you have general advice then maybe you could add some details here https://github.com/grafana/grafana/blob/master/contribute/style-guides/e2e.md

I had some problems with reading multiple values before navigating to the settings page. It was also the same issue when collecting the values after returning to the dashboard again.

But after going through the cypress documentation, I realized that cypress first run the entire flow (without executing anything) just to figure out order of things. After the order is decided everything is being executed accordingly. Then I managed to apply the right changes to make this work as I wanted.

Before that I tried some async/await things :P

**Pull request merged**